plan/logical_plan_builder.go

```diff
@@ -1426,19 +1452,52 @@ func (b *planBuilder) buildUpdateLists(tableList []*ast.TableName, list []*ast.A
 				b.err = ErrBadGeneratedColumn.GenByArgs(colInfo.Name.O, tableInfo.Name.O)
 				return nil, nil
 			}
+			if asNames, ok := tableAsName[tableInfo]; ok {
+				// NOTE: "UPDATE t m, t n SET m.a=m.a+10, n.b=n.b+10" won't set both t.a=t.a+10 and t.b=t.b+10.
+				// But "UPDATE t m, t n SET m.a=m.a+10, m.b=m.b+10" can do that.
+				// So here, we change generated columns in all alias in order to ensure we can do it correctly.
+				for _, asName := range asNames {
+					virtualAssignments = append(virtualAssignments, &ast.Assignment{
+						Column: &ast.ColumnName{Table: asName, Name: colInfo.Name},
```

Comment:
We don't need. Because this is for this case:
```sql
update t1 AS m, t2 AS n, ...
```
the alias after `AS`, **m**, **n**, ...., can't have schema name.